apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -77,6 +66,204 @@ mod prepare_block {
                 ..Default::default()
             }
         }
+
+        /// Builds a batch of vote extension transactions, comprised of Ethereum
+        /// events
+        fn build_vote_extensions_txs(
+            &mut self,
+            local_last_commit: Option<ExtendedCommitInfo>,
+        ) -> Vec<TxRecord> {
+            let protocol_key = self
+                .mode
+                .get_protocol_key()
+                .expect("Validators should always have a protocol key");
+
+            let vote_extension_digest =
+                local_last_commit.and_then(|local_last_commit| {
+                    let votes = local_last_commit.votes;
+                    self.compress_vote_extensions(votes)
+                });
+            let vote_extension_digest =
+                match (vote_extension_digest, self.storage.last_height) {
+                    // handle genesis block
+                    (None, BlockHeight(0)) => return vec![],
+                    (Some(_), BlockHeight(0)) => {
+                        tracing::error!(
+                            "The genesis block should not contain vote \
+                             extensions"
+                        );
+                        // TODO: maybe slash validators who claim to have
+                        // seen vote extensions at H=0
+                        return vec![];
```

Comment:
this scenario is possible if byzantine validators claim they have vote extensions to act upon, when the last height is 0 (ie. the genesis block). since vote extensions are included at the end of a round in the precommit, they couldn't have seen any vote extensions then.

Reply:
When the last height is 0, there has been no previous round (or previous precommit stage), so there definitely shouldn't be any vote extensions? ie byzantine validators haven't had a chance to call `ExtendVote` yet. I would have thought Tendermint could never be handing us vote extensions during `PrepareProposal` for the genesis block. i.e. `self.storage.last_height == 0` -> `req.local_last_commit.votes.is_empty()` as block proposer.

Reply:
ah actually good point, but I don't think anything else gets called at height 0. definitely worth double checking though.

Reply:
I don't see anything related with this in the [specs](https://docs.tendermint.com/master/spec/abci++/abci++_methods.html), so I'm marking the issue as resolved